File: 15주차/SG/Path_Sum.cpp
class Solution
{
public:
    bool pathSum(TreeNode *cur,int curSum,int targetSum){
        if(cur == NULL) return false;
        curSum += cur->val;
        if(cur->left == NULL && cur->right == NULL){  
            if(curSum == targetSum) return true;
            return false;
        }
        return pathSum(cur->left,curSum,targetSum) | pathSum(cur->right,curSum,targetSum);
    }
    bool hasPathSum(TreeNode *root, int targetSum)
    {
        if(root == NULL) return false;
        return pathSum(root,0,targetSum);
    }
};

File: 15주차/NH/Path Sum.js
function hasPathSum(root, targetSum) {
    let answer=[];
    
    function sample(root,sum){
    if(!root)return false;
     
    let left=root.left?sample(root.left,sum+root.val):sum+root.val;
    let right=root.right?sample(root.right,sum+root.val):sum+root.val;
   
   if((targetSum==left&&typeof(left)=='number'&&!root.right)||(targetSum==right&&typeof(right)=='number'&&!root.left)){
        return true;
    }
    if(((typeof(left)=='boolean')&&(left==true))||((typeof(right)=='boolean')&&(right==true))){   
        return true;
    }
    else return false;
    }
    return sample(root,0)
};